### PR TITLE
merge bitcoin#16566...#18004: auxillary backports

### PR DESCRIPTION
**Pull request by kittywhiskers**
## Additional Notes

* This PR aims to close https://github.com/bitcoin/bitcoin/pull/17197 and https://github.com/bitcoin/bitcoin/pull/18218 by merging every PR within the meta-PR. 
  
  Most PRs were already merged thanks to backporting efforts, whatever that remains has been included in this PR; those PRs are...
  * bitcoin#17095: Filter control characters out of log messages 
  * bitcoin#18004: don't embed a build-id when building libdmg-hfsplus

* PRs that don't really have a theme have also been included here

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/qt/clientmodel.cpp [hunk 1]
```diff
@@ -40,23 +41,37 @@ ClientModel::ClientModel(interfaces::Node& node, OptionsModel *_optionsModel, QO
...
-    pollTimer->start(MODEL_UPDATE_DELAY);
     mnListCached = std::make_shared<CDeterministicMNList>();
 
+    QTimer* timer = new QTimer;
+    timer->setInterval(MODEL_UPDATE_DELAY);
+    connect(timer, &QTimer::timeout, [this] {
+        // no locking required at this point
+        // the following calls will acquire the required lock
+        Q_EMIT mempoolSizeChanged(m_node.getMempoolSize(), m_node.getMempoolDynamicUsage());
+        Q_EMIT islockCountChanged(m_node.llmq().getInstantSentLockCount());
```
[the remark this answers is not in the document]
Comment (kittywhiskers) [reply src/qt/clientmodel.cpp:58]: * `bytesChanged` was introduced in bitcoin#2924 and was removed in https://github.com/dashpay/dash/commit/8a6f55f6a5a18e01048ccac56f637291d87ba0e9 (https://github.com/dashpay/dash/pull/3720)
* Nice catch on #17427, backported!

File: src/qt/clientmodel.cpp [hunk 2]
```diff
@@ -29,6 +29,7 @@
 #include <stdint.h>
 
 #include <QDebug>
+#include <QThread>
```
Comment (PastaPastaPasta) [inline src/qt/clientmodel.cpp:32]: Do NOT backport 17252, please backport 17135 instead! (note that the diff is the same)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
LGTM, just one issue

File: src/qt/clientmodel.cpp (hunk 2 above)
Comment (kittywhiskers) [reply src/qt/clientmodel.cpp:32]: Renamed commit to reference 17135 instead, force pushed!

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**